### PR TITLE
Target date default to Now #7711

### PR DESCRIPTION
**Pull request by elvisdorkenoo**
# Description

This PR changes the target emitter to default to now when the config target date is null or is a function that returns null.

medic/cht-conf#516

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or integration tests where appropriate
- Backwards compatible: Works with existing data and configuration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: test/nools/target-emitter.spec.js [hunk 1]
```diff
@@ -149,6 +149,27 @@ describe('target emitter', () => {
...
+          targets: [ target ],
+          tasks: [],
+        };
+
+        // when
+        const emitted = runNoolsLib(config).emitted;
+
+        // then
+        assert.deepEqual(emitted, [
+          { _id: 'c-2~pT-1', _type:'target', date:TEST_DATE },
```
Comment (latin-panda) [inline test/nools/target-emitter.spec.js:168]:
```suggestion
          { _id: 'c-2~pT-1', _type: 'target', date: TEST_DATE },
```
Please add these spaces between property and value

File: test/nools/target-emitter.spec.js [hunk 2]
```diff
@@ -149,6 +149,27 @@ describe('target emitter', () => {
...
+          { _type:'_complete', _id: true },
```
Comment (latin-panda) [inline test/nools/target-emitter.spec.js:169]:
```suggestion
          { _type: '_complete', _id: true },
```
Also this little one, please

File: test/nools/target-emitter.spec.js [hunk 3]
```diff
@@ -149,6 +149,27 @@ describe('target emitter', () => {
         ]);
       });
 
+      it('should default to "now" if target date is a function returning null', () => {
```
Comment (latin-panda) [inline test/nools/target-emitter.spec.js:152]: Please add a test for the case when date is assigned null,  `target.date = null;`

**Review by latin-panda — CHANGES_REQUESTED** [review]
Looking good, just a couple of request inline please.
Thanks!

File: test/nools/target-emitter.spec.js (hunk 3 above)
Comment (elvisdorkenoo) [reply test/nools/target-emitter.spec.js:152]: I have tried to add it and got an error, then tried to upload a setting config with a `target.date = null`, cht-conf validations step rejected if, that's why I did not add that test. 
image

Comment (latin-panda) [reply test/nools/target-emitter.spec.js:152]: Do you get the same if you assign explicitly `undefined`?

Comment (elvisdorkenoo) [reply test/nools/target-emitter.spec.js:152]: `undefined` passes the validation...

Comment (latin-panda) [reply test/nools/target-emitter.spec.js:152]: Okay, but I think that case is already covered below, right?

Comment (elvisdorkenoo) [reply test/nools/target-emitter.spec.js:152]: I think so, it's undefined if no value is set, but I can add an unit test where I explicitly set the date to undefined, if you find it useful.

Comment (latin-panda) [reply test/nools/target-emitter.spec.js:152]: No, I think like this is okay
